[FEAT] 타인 프로필 보기에서 풀이 ID도 주도록 수정

jd99iam opened the pull request:
## #199 

### 💡 PR 타입(하나 이상의 PR 타입을 선택해주세요)
* [ ] 기능 추가
* [ ] 기능 삭제
* [ ] 버그 수정
* [ ] 테스트코드 작성
* [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 📲 반영 브랜치
feat/199-yujinFix->develop

### ❔ 변경 사항
타인 프로필 보기에서 풀이 ID도 주도록 수정

### 🔍 테스트 결과
- Repo 테스트
- Service 테스트
- Controler 테스트

### 📰 참고 Reference

### ⚒️ 기타 사항

jd99iam committed:
Merge pull request #198 from rlfrkdms1/develop

[FEAT] 프로필 풀이에 언어 추가

jd99iam committed:
refactor: 정적 팩터리 메서드

정적 팩터리 메서드 리팩터링
- 인자 명 target->solution으로 수정
- 정적 팩터리 메서드 from으로 rename

jd99iam committed:
refactor: 정적 팩터리 메서드

정적 팩터리 메서드 리팩터링
- NPE 예방을 위해 수정

jd99iam committed:
refactor: 미사용 import문 삭제

jd99iam committed:
feat: 타인 프로필 보기 dto 수정

풀이 dto에서 풀이 id도 주도록 수정

related to: #199

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/member/response/SolutionWithProblemResponse.java:
```diff
@@ -10,6 +10,7 @@
 public class SolutionWithProblemResponse {
 
     private String problemName;
+    private Long id;
```
이렇게 하면 solution id인지 확인하기 어렵지 않나요?

rlfrkdms1 replied:
차라리 solution이랑 problem을 나누거나 하는게 나을 것 같아요

jd99iam replied:
> 차라리 solution이랑 problem을 나누거나 하는게 나을 것 같아요

풀이를 위한 dto 따로 만들어서 묶고 거기에 title 추가했습니다. problemName은 problemTitle로 rename했습니다

rlfrkdms1 commented:
commit의 세부 내역을 적어주신 것은 좋습니다 !
그런데 커밋의 제목이라고 할 수 있는 subject가 너무 부실한 것 같아요 !
commit 내용의 요약이 되는 제목을 적어주시면 좋을 것 같습니다~

jd99iam committed:
feat: 타인 프로필 보기 문제, 풀이 응답 dto 수정

타인 프로필 보기 문제, 풀이 응답 dto 수정
- 문제 제목 필드 problemName -> problemTitle로 rename
- 풀이 관련 필드를 SolutionForProfileResponse로 묶음
- SolutionForProfileResponse에 풀이 제목 title 필드 추가

jd99iam commented:
> commit의 세부 내역을 적어주신 것은 좋습니다 ! 그런데 커밋의 제목이라고 할 수 있는 subject가 너무 부실한 것 같아요 ! commit 내용의 요약이 되는 제목을 적어주시면 좋을 것 같습니다~

네 커밋 메세지 더 신경 쓰겠씁니다 감사합니다!!

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/solution/response/SolutionForProfileResponse.java:
```diff
@@ -0,0 +1,33 @@
+@Getter
+@Builder
+public class SolutionForProfileResponse {
+    private static final String REGEX = "\n";
```
`System.lineSeparator()` 사용하면 좋을 것 같습니다

jd99iam replied:
> `System.lineSeparator()` 사용하면 좋을 것 같습니다

수정하겠습니다

rlfrkdms1 commented on backend/src/main/java/com/prograngers/backend/dto/solution/response/SolutionForProfileResponse.java:
```diff
@@ -0,0 +1,33 @@
+                .id(solution.getId())
+                .dataStructure(solution.getDataStructureView())
+                .algorithm(solution.getAlgorithmView())
+                .language(solution.getLanguage().getView())
```
음 .. NPE를 고려한게 맞을까요 ?

jd99iam replied:
> 음 .. NPE를 고려한게 맞을까요 ?

language는 필수값이라 괜찮다고 생각했는데 추후 변경을 고려해서 메서드 추가하겠습니다

rlfrkdms1 replied:
아하 그렇네요 일단 이렇게 해도 괜찮을 것 같아요

jd99iam replied:
> 아하 그렇네요 일단 이렇게 해도 괜찮을 것 같아요

일단 getLanguageView() 추가해서 해당 dto에만 적용했어요!

rlfrkdms1 replied:
음 .. 사실 Language에는 컨버터가 없어서 getLanguageView() 메서드의 추가가 의미없을 것 같아요

jd99iam committed:
refactor: 개행문자 변경

운영체제에 알맞은 개행문자를 사용할 수 있도록 System.lineSeparator() 로 변경

jd99iam committed:
feat(Solution): getLanguageView 추가

언어값이 null 일 경우를 대비해 getLanguageView 추가
- 언어값이 null이면 soltuion.getLanguage().getView()가 NPE가 발생함
- getLanguageView에서 null 검사를 하고 반환하도록 추가

rlfrkdms1 commented:
그리고 브랜치 이름 작명이 지금 feat/199-yujinFix인데 적절한 작명이 아닌 것 같습니다. 
누가 요청했는지가 아니라 어떤 수정을 요청했는지 어떤 수정이 이루어질 것인지를 적는게 좋을 것 같아요

jd99iam commented:
> 그리고 브랜치 이름 작명이 지금 feat/199-yujinFix인데 적절한 작명이 아닌 것 같습니다. 누가 요청했는지가 아니라 어떤 수정을 요청했는지 어떤 수정이 이루어질 것인지를 적는게 좋을 것 같아요

저도 원래 수정사항에 대해 브랜치를 작명하려고 했었는데 수정사항이 관련 없는 두군데에 있는 것이라 어떤식으로 작명할지 고민하다가 이렇게 했네요. 그럼 여러 수정사항이 있는 경우 브랜치를 여러 개 파는게 맞을까요?

jd99iam committed:
refactor(SolutionForProfileResponse): getLanguageView 사용하도록 변경

Language가 추후 nullable로 바뀌는 변경에 대비한 수정
- getLanguage().getView() -> getLanguageView()로 수정

rlfrkdms1 commented:
음,, 지금 PR 제목이 `타인 프로필 보기에서 풀이 ID도 주도록 수정`이니까 feature/profile-fix 등의 방법이 있을 것 같아요 굳이 엔티티명이 오지 않아도 될 것 같습니다.

jd99iam commented:
> 음,, 지금 PR 제목이 `타인 프로필 보기에서 풀이 ID도 주도록 수정`이니까 feature/profile-fix 등의 방법이 있을 것 같아요 굳이 엔티티명이 오지 않아도 될 것 같습니다.

넵 일단 수정사항들 반영해서 push했습니다 추가 수정사항 있으면 comment 해 주세요. approve 해 주시면 merge 하고 배포하겠습니다

jd99iam committed:
feat(Solution): GetLanguageView 삭제

Language는 컨버터가 없어서 getLanguageView가 불필요해서 삭제
- Solution에서 메서드 삭제
- SolutionForProfileResponse에서 사용하지 않도록 변경
